internal/txapp/txapp.go

```diff
@@ -205,6 +208,33 @@ func (r *TxApp) ChainInfo(ctx context.Context) (int64, []byte, error) {
 	return height, appHash, nil
 }
 
+// ReloadDB reloads the database state into the engine.
+func (r *TxApp) ReloadDB(ctx context.Context) error {
+	tx, err := r.Database.BeginReadTx(ctx)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback(ctx)
+
+	// Check if we are switching from StateSync?
+	// If so, we need to reload the engine.
+	height, app_hash, err := getChainState(ctx, tx)
+	if err != nil {
+		return err
+	}
+
+	if r.height == 0 && height > r.height && r.appHash == nil {
+		err = r.Engine.Reload(ctx, tx)
```

Comment:
This is a bit confusing to me. A method called `ReloadDB` that reloads our in-memory catalog that is actually dependent on the current state of the chain. I'm not totally sure what the point of this logic is, but given how not often this method would be called, why wouldn't we just call `r.Engine.Reload` every time?

This might lead to us table scanning the database catalog unnecessarily on startup, but I feel like that is not a very big deal, and makes this function better do what it says it does (which makes it less bug prone)

If this logic is really necessary, we should get a better function name.

Reply:
Ah yeah, I had this logic earlier to determine if Info is getting called from Statesync or normal block sync init. Now that we moved it out of Info() we don't these checks, I will remove them